openedx/core/djangoapps/enrollments/views.py

```diff
@@ -699,9 +699,23 @@ def post(self, request):
 
         has_api_key_permissions = self.has_api_key_permissions(request)
 
-        # Check that the user specified is either the same user, or this is a server-to-server request.
+        # Check if the user or email was defined.
         if not username:
-            username = request.user.username
+            email = request.data.get('email')
+            if email:
+                try:
+                    username = User.objects.get(email=email).username
+                except ObjectDoesNotExist:
+                    return Response(
+                        status=status.HTTP_406_NOT_ACCEPTABLE,
+                        data={
+                            'message': f'The user with the email address {email} does not exist.'
+                        }
+                    )
```

Comment:
This seems to reveal to any authenticated user whether an email exists in the system? Would love your opinions on the security risk. I feel this could depend on:
- Whether `email` is intended to be for only server-to-server calls? Might be easy to just move this under the permission check below to make this intention explicit.

- Whether the `EnrollmentUserThrottle` on this endpoint is good enough to prevent any abuse. It might already be! We do reveal the existence of registered emails already on our password reset endpoint [[GitHub](https://github.com/openedx/edx-platform/blob/2ffcc1aa3fe3ac37272d36ad0f0ae495f66f0804/openedx/core/djangoapps/user_authn/views/password_reset.py#L284)], but haven't looked too deeply on whether the throttles provide equivalent protection.

Reply:
Excellent that you noticed it. I certainly did not.

For the use case of woocommerce moving this below the server-to-server check would be perfectly fine.
Even if `EnrollmentUserThrottle` works to prevent abuse I think the less endpoints that might reveal this information, the better.

Reply:
Thank @pshiu, for this catch. I'll try put that condition inside the server-to-server check and make the changes.